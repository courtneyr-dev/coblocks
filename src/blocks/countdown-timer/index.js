/**
 * External dependencies
 */
import { TimerIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies.
 */
import edit from './edit';
// import example from './example';
import metadata from './block.json';
import save from './save';

/**
 * WordPress dependencies.
 */
import { Icon } from '@wordpress/components';
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	description: __( 'Add a countdown timer to promote limited time offers or upcoming events.', 'coblocks' ),
	edit,
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'countdown', 'coblocks' ),
		/* translators: block keyword */
		__( 'timer', 'coblocks' ),
	],
	save,
	supports: {
		align: [ 'wide', 'full' ],
		multiple: false,
	},
	title: _x( 'Countdown Timer', 'block name', 'coblocks' ),
};

export { metadata, name, category, settings };
