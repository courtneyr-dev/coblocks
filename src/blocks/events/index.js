/**
 * External dependencies
 */
import { EventsIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies.
 */
import edit from './edit';
import example from './example';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import deprecated from './deprecated';

/**
 * WordPress dependencies.
 */
import { __, _x } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	title: _x( 'Events', 'block name', 'coblocks' ),
	description: __( 'Add a list of events or display events from a public calendar.', 'coblocks' ),
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'calendar', 'coblocks' ),
		/* translators: block keyword */
		__( 'date', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
	},
	// example,
	transforms,
	edit,
	save,
	deprecated
};

export { metadata, name, category, settings };
