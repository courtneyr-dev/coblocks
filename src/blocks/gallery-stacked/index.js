/**
 * External dependencies
 */
import { GalleryStackedIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import { BLOCK_VARIATION_GALLERY_STACKED } from '../../block-variations/core/gallery';
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { dispatch } from '@wordpress/data';
import { createBlock, switchToBlockType } from '@wordpress/blocks';

/**
 * Block constants
 */
const { name, category } = metadata;

const settings = {
	attributes: {
		ids: { type: 'array' },
		images: { type: 'array' },
		// linkTo: { type: 'string' },
		// sizeSlug: { type: 'string' },
	},
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );
		replaceBlocks(
			[ props.clientId ],
			switchToBlockType( props, 'core/gallery' )
		);
		return null;
	},
	parent: [],
	save: () => null,
	transforms: {
		to: [
			{
				blocks: [ 'core/gallery' ],
				transform: ( attributes ) => {
					return createBlock(
						'core/gallery',
						Object.assign( {}, attributes, BLOCK_VARIATION_GALLERY_STACKED.attributes )
					);
				},
				type: 'block',
			},
		],
	},
};

export { name, category, icon, metadata, settings };
