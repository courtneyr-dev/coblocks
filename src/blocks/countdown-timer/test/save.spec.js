/**
 * External dependencies
 */
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

let block;

describe( name, () => {
	beforeAll( () => {
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		block = createBlock( name );
	} );

	it( 'should render with content', () => {
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
