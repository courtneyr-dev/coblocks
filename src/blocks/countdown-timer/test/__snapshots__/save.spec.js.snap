// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/countdown-timer should render with content 1`] = `
"<!-- wp:coblocks/countdown-timer -->
<div class=\\"wp-block-coblocks-countdown-timer\\">This is the save Countdown</div>
<!-- /wp:coblocks/countdown-timer -->"
`;
